### PR TITLE
feat(app): Display reachable non connectable robots

### LINKED ISSUE #2345: Robots List (Connectivity): Display Reachable Robots
As a Run App user, I would like to be able to see and interact with robots that are reachable, even if I cannot connect to them.

#### Robots that are reachable are displayed in the Robots List Sidebar
- [ ] Card is **not** greyed out
- [ ] Card has an alert icon
- [ ] Card should **not** display pipettes and modules
- [ ] Card should have a right chevron in place of the connection toggle

#### Robot Details Page should display a modified set of information and affordances:
- [ ] Update button is available if update server is up
- [ ] Information panel is available if update server is up
  - [ ] Information panel includes robot name, server version, and firmware version (current behavior)
  - Note we have access to but do not display API server version and System/Container version
- [ ] All other cards and affordances are greyed out / inactive

#### User should see a warning explaining their robot's connectivity state over the Robot Details Page 
- [ ] Copy TBD 
- [ ] Copy is different depending on whether:
  1. Robot is advertising but cannot boot any of its servers 
  2. Robot is advertising and can only boot update server

## Design and Copy

- https://zpl.io/aMvXMgJ
- https://zpl.io/25rjnDe

- warning copy: TBD

## Implementation Notes
- This scenario is possible in two cases:
  1. Robot is advertising but cannot boot any of its servers 
  2. Robot is advertising and can only boot update server
- Note we do know the connection method

### PR DESCRIPTION
**Pull request by Kadee80**
## overview

closes #2345 

## changelog

- feat(app): Display reachable non connectable robots

## review requests

To begin, SSH into a robot and kill the main server.

### `ok: false, serverOk: false, advertising: true`

SSH into a robot and kill the update server, click refresh in the app, then check:

- [ ] robot displays as reachable in list (no connection toggle)
- [ ] all cards _except_ Robot `InformationCard` are disabled
- [ ] update button in `InformationCard` is disabled
- [ ] alert banner message  = 'Robot is advertising but cannot boot any of its servers.'

### `ok: false, serverOk: true, advertising: X`

This test is the same whether or not the robot is advertising. To test without the robot advertising, SSH into the robot and kill `announce_mdns.py`

- [ ] robot displays as reachable in list (no connection toggle)
- [ ] all cards _except_ Robot `InformationCard` are disabled
- [ ] update button in `InformationCard` is still enabled
- [ ] alert banner message  = 'Robot is advertising and can only boot update server'

### Question:

~Do we want to sneak in remove `opentrons-` prefix here?~ Nope, followup PR

### CODE REVIEW

File: app/src/components/RobotSettings/AdvancedSettingsCard.js [hunk 1]
```diff
@@ -60,58 +66,66 @@ class BooleanSettingToggle extends React.Component<BooleanSettingProps> {
 }
 
 function AdvancedSettingsCard (props: Props) {
-  const {name, settings, set, fetch, download, health} = props
-  const logsAvailable = health && health.response && health.response.logs
+  const {settings, set, fetch, download} = props
+  const {name, health, status} = props.robot
+  const disabled = status !== CONNECTABLE
+  const logsAvailable = health && health.logs
   const resetUrl = `/robots/${name}/reset`
```
Comment (mcous) [inline app/src/components/RobotSettings/AdvancedSettingsCard.js]: Should be passed down via props

File: app/src/components/RobotSettings/AdvancedSettingsCard.js [hunk 2]
```diff
@@ -4,29 +4,35 @@ import * as React from 'react'
...
-type DP = {
+type DP = {|
   fetch: () => mixed,
   set: (id: string, value: boolean) => mixed,
   download: () => mixed,
-}
+|}
 
-type Props = OP & SP & DP
+type Props = {
```
Comment (mcous) [inline app/src/components/RobotSettings/AdvancedSettingsCard.js]: Style nitpick for self: can be a one-liner

File: app/src/components/RobotSettings/ConnectivityCard.js [hunk 3]
```diff
@@ -20,23 +20,27 @@ import WifiConnectForm from './WifiConnectForm'
 import WifiConnectModal from './WifiConnectModal'
 
 import type {State, Dispatch} from '../../types'
-import type {Robot} from '../../robot'
+import type {ViewableRobot} from '../../discovery'
 
-type OwnProps = Robot
+type OwnProps = {robot: ViewableRobot}
```
Comment (mcous) [inline app/src/components/RobotSettings/ConnectivityCard.js]: Style nitpick for self: `OP`, `DP`, `SP`

File: app/src/pages/Robots/RobotSettings.js [hunk 4]
```diff
@@ -36,19 +39,27 @@ type SP = {
...
-  robot: Robot,
+  robot: ViewableRobot,
   match: Match,
 }
 
-type Props = SP & OP & {
-  closeHomeAlert?: () => mixed,
-  closeConnectAlert: () => mixed,
-}
+type Props = SP &
```
Comment (mcous) [inline app/src/pages/Robots/RobotSettings.js]: Switch to spread

File: app/src/pages/Robots/index.js [hunk 5]
```diff
@@ -4,19 +4,21 @@ import * as React from 'react'
...
+import {selectors as robotSelectors} from '../../robot'
+import {CONNECTABLE, getConnectableRobots, getReachableRobots} from '../../discovery'
 
 import {Splash} from '@opentrons/components'
 import Page from '../../components/Page'
 import RobotSettings from './RobotSettings'
 import InstrumentSettings from './InstrumentSettings'
 
+import type {State} from '../../types'
+// import type {Robot} from '../../robot'
```
Comment (mcous) [inline app/src/pages/Robots/index.js]: Can delete this line

File: app/src/robot/types.js [hunk 6]
```diff
@@ -29,18 +29,21 @@ export type Slot =
...
   name: string,
 }
 
+// TODO(mc, 2018-10-09): deprecate this type
 // robot MDNS service for connectivity
 export type RobotService = BaseRobot & {
   ip: string,
   port: number,
-  wired: ?boolean,
+  // wired: ?boolean,
```
Comment (mcous) [inline app/src/robot/types.js]: Can delete this line

File: components/src/nav/SidePanel.css [hunk 7]
```diff
@@ -9,7 +9,6 @@
   overflow: hidden;
   border-right: var(--bd-light);
   background-color: var(--c-bg-light);
-  transition: all 0.5s ease-in-out;
```
Comment (mcous) [inline components/src/nav/SidePanel.css]: @IanLondon or @b-cooper this change should affect PD, right? It's a artifact of a time when the side-panel was closable

File: discovery-client/src/index.js [hunk 8]
```diff
@@ -64,7 +64,7 @@ export default function DiscoveryClientFactory (options?: Options) {
 export const SERVICE_EVENT: 'service' = 'service'
 export const SERVICE_REMOVED_EVENT: 'serviceRemoved' = 'serviceRemoved'
 export const DEFAULT_POLL_INTERVAL = 5000
-export const DEFAULT_DISCOVERY_INTERVAL = 60000
+export const DEFAULT_DISCOVERY_INTERVAL = 90000
```
Comment (mcous) [inline discovery-client/src/index.js]: During testing, 60 seconds didn't seem like enough time to gather all advertisments. Hopefully we can be more nuanced in the future with how we determine a robot has stopped advertising, but right now we're a little limited by our mDNS library

File: components/src/nav/SidePanel.css (hunk 7 above)
Comment (b-cooper) [reply components/src/nav/SidePanel.css]: nope, you should be fine

**Pull request merged**